lib/bridge/MatrixHandler.js

```diff
@@ -833,6 +833,25 @@ MatrixHandler.prototype._onJoin = Promise.coroutine(function*(req, event, user)
             let bridgedClient = yield self.ircBridge.getBridgedClient(
                 room.server, user.getId(), (event.content || {}).displayname
             );
+
+            // Check for a displayname change and update nick accordingly.
+            if (event.content.displayname !== bridgedClient.displayName) {
+                bridgedClient.displayName = event.content.displayname;
```

Comment:
Presumably we don't persist this, so every bridge restart followed by `onJoin` will trigger a nick change once for each user who doesn't have an explicit display name set?

Reply:
In that case both `event.content.displayname` and `bridgedClient.displayName` should be `null` because `bridgedClient.displayName`  is initialised as follows:

```JavaScript
this.displayName = matrixUser ? this.matrixUser.getDisplayName() : null;
```

Reply:
SGTM